docs.go

```diff
@@ -0,0 +1,289 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/cpuguy83/go-md2man/md2man"
+)
+
+const markdownTemplateString = `% {{ .App.Name }}(8) {{ .App.Description }}
+% {{ .App.Author }}
+% {{ .Date }}
+
+# NAME
+
+{{ .App.Name }}{{ if .App.Usage }} - {{ .App.Usage }}{{ end }}
+
+# SYNOPSIS
+
+{{ .App.Name }}
+{{ if .SynopsisArgs }}
+` + "```" + `{{ range $v := .SynopsisArgs }}
+{{ $v }}{{ end }}
+` + "```" + `
+{{ end }}{{ if .App.UsageText }}
+# DESCRIPTION
+
+{{ .App.UsageText }}
+{{ end }}
+**Usage**:
+
+` + "```" + `
+{{ .App.Name }} [GLOBAL OPTIONS] command [COMMAND OPTIONS] [ARGUMENTS...]
+` + "```" + `
+{{ if .GlobalArgs }}
+# GLOBAL OPTIONS
+{{ range $v := .GlobalArgs }}
+{{ $v }}{{ end }}
+{{ end }}{{ if .Commands }}
+# COMMANDS
+{{ range $v := .Commands }}
+{{ $v }}{{ end }}{{ end }}`
```

Comment:
⭐⭐ _(seriously consider)_

So this is a big ask, but I wanted to ask it anyways! Can you consider possibly [generating this template from / sharing it with] the main help output template? When markdown generation is its own stand-alone template, people might not realize that it exists.

Reply:
Hm, I moved the templates to `template.go`, made them private and const (breaking change). I could also let them var and public but I saw no real use case for having them public.

I added the markdown template there as well, but I'm not sure if and how we should make them dependent of each other... What do you think?